modules/reindex/src/main/java/org/elasticsearch/index/reindex/RestReindexAction.java

```diff
@@ -147,6 +174,31 @@ protected ReindexRequest buildRequest(RestRequest request) throws IOException {
         }
     }
 
+    private static String extractString(Map<String, Object> source, String name) {
+        Object value = source.remove(name);
```

Comment:
Can you clarify why this needs to be removed from the map instead of just retrieved? Does it break something further down?